src/index.ts

```diff
@@ -60,25 +60,151 @@ const plugin: JupyterFrontEndPlugin<void> = {
     // Adding a data attribute
     widget.node.setAttribute('data-is-filesystem-access', '');
 
-    const openDirectoryButton = new ToolbarButton({
-      icon: folderIcon,
-      onClick: async () => {
-        const directoryHandle = await window.showDirectoryPicker();
-
-        if (directoryHandle) {
-          drive.rootHandle = directoryHandle;
+    toolbarRegistry.registerFactory(
+      'FileBrowser', // Factory name
```

Comment:
@jtpio the `FileBrowser` entry in the toolbars section is to match this name. If here you define a different name, you will be your own master for items that goes in this filebrowser toolbar. But it also means that for example you will not get the `Add Launcher` button as it targets the default `FileBrowser` namespace.

Does it clarify your question?

Reply:
yes thanks! I think what we need here is what is described in https://jupyterlab.readthedocs.io/en/latest/extension/extension_points.html#generic-widget-with-toolbar

Reply:
I'll push some changes in https://github.com/jupyterlab-contrib/jupyterlab-filesystem-access/pull/55.